builtin/mainmenu/dlg_version_info.lua

```diff
@@ -0,0 +1,157 @@
+--[[
+Minetest
+Copyright (C) 2018-2020 SmallJoker
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+]]
+
+local function version_info_formspec(data)
+	local body = {
+		"<tag name=h1 color=#0E0>",
+	"<h1>", fgettext("A new Minetest version is available"), "</h1>\n",
+	fgettext("Current version: $1\nNew version: $2\n" ..
+			"Visit $3 to find out how to get the newest version to stay up to date" ..
+			" with the features and bugfixes.",
+		core.get_version().string, data.new_version, data.url)
+	}
+
+	local fs = {
+		"formspec_version[3]",
+		"size[10,5.5]",
+		"hypertext[0.375,0.375;9.25,3.75;;",
+			core.formspec_escape(table.concat(body, "")), "]",
```

Comment:
As @erlehmann pointed out, `formspec_escape` isn't enough here. You have to consider Hypertext escaping as well. You can also omit the empty string as second `table.concat` parameter.

Reply:
This was intentional, as it allows the backend to include markup. I don't believe there's anything security sensitive in the mark up language, although it could be used to crash the client

Reply:
body contains markup, we don't want to escape that